File: lua/classTest2.lua
Person={}
function Person:new(name, age)
	self.name = name
	self.age = age
	return self
end

function Person:say()
	print(self.name, self.age)
end

Student=Person:new(name, age)
function Student:new(name, age, grade)
	self.name = name
	self.age = age
	self.grade = grade
	return self
end

function Student:say()
	print(self.name, self.age, self.grade)
end

function test()
	a = Student:new("Stanley", 20, 8)
	a:say()
end

File: ruby/classTest2.rb

class Person
	def initialize(name, age)
		@name = name
		@age = age
	end

	def say()
		puts "#{@name} #{@age}"
	end
end

class Student < Person
	def initialize(name, age, grade)
		super(name, age)
		@grade = grade
	end

	def say()
		puts "#{@name} #{@age} #{@grade}"
	end
end

def test()
	a = Student.new("Stanley", 20, 8)
	a.say()
end
